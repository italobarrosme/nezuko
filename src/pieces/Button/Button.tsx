import { cn } from '@/utils'
import { Icon } from '@iconify/react'
import { ButtonHTMLAttributes } from 'react'

export type ButtonProps = {
  label?: string
  onClick: () => void
  className?: string
  icon?: string
} & ButtonHTMLAttributes<HTMLButtonElement>

export const Button = ({
  onClick,
  label,
  className,
  type,
  icon = '',
}: ButtonProps) => {
  return (
    <button
      type={type}
      className={cn(
        className,
        'flex items-center gap-4 whitespace-nowrap rounded bg-brand-primary px-4 py-2 font-bold text-white hover:bg-brand-secondary'
      )}
      onClick={onClick}
    >
      {icon ? <Icon icon={icon} width={24} /> : null}
      <p>{label}</p>
    </button>
  )
}
