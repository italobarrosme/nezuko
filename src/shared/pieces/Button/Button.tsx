import { cn } from '@/utils'
import { Icon } from '@iconify/react'
import { ButtonHTMLAttributes, ReactNode } from 'react'

export type ButtonProps = {
  label?: string
  className?: string
  icon?: string
  children: ReactNode
} & ButtonHTMLAttributes<HTMLButtonElement>

export const Button = ({
  className,
  type,
  icon = '',
  children,
  ...props
}: ButtonProps) => {
  return (
    <button
      type={type}
      className={cn(
        'flex items-center gap-4 whitespace-nowrap rounded bg-brand-primary px-4 py-2 font-bold text-white hover:bg-brand-secondary',
        className
      )}
      {...props}
    >
      {icon ? <Icon icon={icon} width={24} /> : null}
      {children}
    </button>
  )
}
