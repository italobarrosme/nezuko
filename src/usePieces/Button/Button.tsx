import { Icon } from "@iconify/react"
import clsx from "clsx"
import { ButtonHTMLAttributes } from "react"

export type ButtonProps = {
  label?: string
  onClick: () => void
  className?: string
  icon?: string
} & ButtonHTMLAttributes<HTMLButtonElement>

export const Button = ({onClick, label, className, type, icon = ''}: ButtonProps) => {
  return (
    <button
      type={type}
      className={clsx(className, 'bg-brand-primary hover:bg-brand-secondary text-white font-bold py-2 px-4 rounded whitespace-nowrap flex items-center gap-4')}
      onClick={onClick}
    >
      {icon ? <Icon icon={icon} width={24} /> : null}
      <p>{label}</p>
    </button>
  )
}