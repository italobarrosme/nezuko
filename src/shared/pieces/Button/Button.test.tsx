import { render, screen } from '@testing-library/react'
import { Button, ButtonProps } from './Button'

const renderComponent = ({ label, className, icon, onClick }: ButtonProps) =>
  render(
    <Button label={label} className={className} icon={icon} onClick={onClick} />
  )

describe('Button', () => {
  it('should render correctly', () => {
    renderComponent({
      label: 'Test',
      className: 'test',
      icon: 'test',
      onClick: () => jest.fn(),
      type: 'button',
    })

    expect(screen.getByText('Test')).toBeTruthy()
  })

  it('should render correctly background color', () => {
    renderComponent({
      label: 'Test',
      icon: 'test',
      onClick: () => jest.fn(),
      type: 'button',
    })

    const element = screen.getByRole('button')

    expect(element.className).toMatch('bg-brand-primary')
  })

  it('should will snapshot', () => {
    const { container } = renderComponent({
      label: 'Test',
      icon: 'test',
      className: 'test',
      onClick: () => jest.fn(),
      type: 'button',
    })

    expect(container).toMatchSnapshot()
  })
})
