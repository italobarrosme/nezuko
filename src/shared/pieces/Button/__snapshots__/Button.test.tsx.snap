// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should will snapshot 1`] = `
<div>
  <button
    class="test flex items-center gap-4 whitespace-nowrap rounded bg-brand-primary px-4 py-2 font-bold text-white hover:bg-brand-secondary"
  >
    <span />
    <p>
      Test
    </p>
  </button>
</div>
`;
